snap/info.go

```diff
@@ -187,16 +187,26 @@ func (s *Info) CommonDataDir() string {
 	return filepath.Join(dirs.SnapDataDir, s.Name(), "common")
 }
 
-// DataHomeDir returns the per user data directory of the snap.
-func (s *Info) DataHomeDir() string {
+// DataHomeGlob returns the per user data directory of the snap.
+func (s *Info) DataHomeGlob() string {
 	return filepath.Join(dirs.SnapDataHomeGlob, s.Name(), s.Revision.String())
 }
 
-// CommonDataHomeDir returns the per user data directory common across revisions of the snap.
-func (s *Info) CommonDataHomeDir() string {
+// RootUserDataDir returns the user data dir for the root user
+func (s *Info) RootUserDataDir() string {
```

Comment:
There's no reason for this to be special in this API. Every single user has a home data dir which depends on the username. This can easily be:
```
func (s *Info) UserDataDir(user string) string { ... }